package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import { FlatList, TouchableOpacity, TouchableOpacityProps } from 'react-native';
+import {
+  isSuggestionCommand,
+  isSuggestionEmoji,
+  isSuggestionUser,
+  Suggestion,
+  SuggestionComponentType,
+  Suggestions,
+  useSuggestionsContext,
+} from '../../contexts/suggestionsContext/SuggestionsContext';
+import type { DefaultUserType } from '../../types/types';
+
+import type { AutoCompleteSuggestionHeaderProps } from './AutoCompleteSuggestionHeader';
+import type { AutoCompleteSuggestionItemProps } from './AutoCompleteSuggestionItem';
+import type { SuggestionsContextValue } from '../../contexts/suggestionsContext/SuggestionsContext';
+import { useTheme } from '../../contexts/themeContext/ThemeContext';
+
+type AutoCompleteSuggestionListComponentProps<Us extends DefaultUserType = DefaultUserType> = {
```

Comment:
Yes, the props are different from the actual props, so to differentiate I kept it this way. Any better suggestions are welcome, just make sure you go through how this is used I think it will give you clarity about the usage. :)

Reply:
I think this should be called `AutoCompleteSuggestionListProps` to conform with how it's done everywhere else in our code, and the added props from other type definitions should be raised to this type definition. Meaning that it'd look like this:

```tsx
type AutoCompleteSuggestionListComponentProps<Us extends DefaultUserType = DefaultUserType> = {
  active: boolean;
  headerProps: AutoCompleteSuggestionHeaderProps;
  suggestions: Suggestions<Us>;
  type: SuggestionComponentType;
  AutoCompleteSuggestionHeader?: React.ComponentType<AutoCompleteSuggestionHeaderProps>;
  AutoCompleteSuggestionItem?: React.ComponentType<AutoCompleteSuggestionItemProps<Us>>;
} & Pick<SuggestionsContextValue, 'AutoCompleteSuggestionHeader' | 'AutoCompleteSuggestionItem'>;
```
Then `AutoCompleteSuggestionListPropsWithContext` and the current `AutoCompleteSuggestionListProps` types can be removed rather than having the same thing defined three times, and I think is clearer as the props don't deal with a context - the component `AutoCompleteSuggestionListWithContext` does. If this component needed a different set of props than the export of `AutoCompleteSuggestionList` (which probably would be a sign that something should be changed), it could have a Prop suffix to what's passed in there, but luckily it's the same props as the exported member that wraps the memoized component.

Reply:
Please see the `MessageDeleted` component as a reference. I have followed a similar architecture :)